mlflow/tracing/processor/mlflow.py

```diff
@@ -58,13 +59,22 @@ def on_start(self, span: OTelSpan, parent_context: Optional[Context] = None):
             request_id = trace_info.request_id
         span.set_attribute(SpanAttributeKey.REQUEST_ID, json.dumps(request_id))
 
+        # NB: This is a workaround to exclude the latency of backend StartTrace API call (within
+        #   _create_trace_info()) from the execution time of the span. The API call takes ~1 sec
+        #   and significantly skews the span duration.
```

Comment:
I think ideal solution should be:
 making time-consuming backend invocations like `self._call_endpoint(StartTrace, req_body)` to be async,
 to prevent it influencing client side tracing timestamp

Reply:
Unfortunately it cannot be async, as we rely on the `request_id ` in the response.

Reply:
Curious why can't we generate `request_id` in client side and then send the `request_id` to server.